### ISSUE TITLE
A new cop that discourages the use of specific ruby syntax features

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#12072]
## Is your feature request related to a problem? Please describe.

Ruby is a sweet language, but sometimes is too sweet. Personally, I do not like some ruby syntax sugar features (or do not find them better than the alternatives). 
For example, `unless` keyword, pattern matching, numbered parameters (`_1`, `_2` etc), argument forwarding (`def foo(...)`), one-line method definitions (`def foo = 2`), shorthand hash construction (`{ foo:, bar: }`) and more.   

## Describe the solution you'd like

I would like to have a cop that allows me to configure something like `DisabledSyntax: [unless, numbered_arguments]` that produces an offense when such feature is used. It should be disabled by default with all features enabled.

## Describe alternatives you've considered

I understand that this request is kinda controversial and maybe some people would decide that it works against ruby feature additions. So I can create it as an extension to rubocop. Just wanted to make sure if this is something you wanted to have in rubocop itself.

**Comment by Earlopain** [comment]
For hashes there is

```yml
Style/HashSyntax: 
   EnforcedShorthandSyntax: never
```

I do feel that some of the features of ruby are a bit much but have no issues with forwarding for example. The newly added `Style/ArgumentsForwarding` has some options but none to discourage `...` I believe, perhaps that could be extended. Endless methods and numbered parameters on the other hand would be nothing I would ever want to write or merge myself.

**Comment by koic** [comment]
It looks better to provide an extension gem that restricts unfavorite syntax instead of putting it in the core feature. As for feature restrictions, some options are already provided in individual cops. Thank you.
